ui/helpers/errorMessage.js

```diff
@@ -43,3 +45,25 @@ export const formatTaskclusterError = function formatTaskclusterError(e) {
 
   return `${TC_ERROR_PREFIX}${errorMessage}`;
 };
+
+export const processErrorMessage = function processErrorMessage(
```

Comment:
It's hard to tell what this method does at first glance. From the `503` reference I'm presuming it wraps HTTP requests? If so, this I think this would be best off in the `http` helpers instead?

Reply:
This was an attempt to ensure we always provide a meaningful error message to the user and aggregate them so multiple error messages can be shown at once. So we can receive json responses that look like this:

`{"startday": [
        "This field is required."
    ]}`
So in this case, it'll look up the 'startday' key in prettyErrorMessages and use that message instead.

Or an error message that looks like this:
`{
    "tree": [
        "moz does not exist."
    ]
}`
In this case, we'll return this message since it'd be useful to show the user they entered in the wrong tree (repo) query param

Or in the case of the 503:
[503: "Service Unavailable"] from the `getData` method we'll return a less cryptic message instead.

Looking at it now, it could be simplified with the aggregation part of it moved into the `ErrorMessages` component.

Maybe we should have a discussion about how we want to handle error messaging for TH in general, since error message quality varies depending on whether we receive a meaningful json response or any json response at all?

Reply:
In the end, I decided to clean up `processErrorMessage` to keep it simple - was debating whether to include a `customErrorMessage` parameter but since I wouldn't actually be using it now I can always add that in later.